### ISSUE TITLE
[Feature] Dynamic max building level

### ISSUE THREAD

**Issue by Raycoms** [ldtteam/minecolonies#191]
Should search the styles folder and look up how many of the current style there are.

**Comment by cltnschlosser** [comment]
But we can't have different, max caps for a building. We just need to limit what styles you can have at a given level.

**Comment by Raycoms** [comment]
Why can'T we?

2016-10-09 22:39 GMT-03:00 Colton Schlosser <notifications@github.com>:

> But we can't have different, max caps for a building. We just need to
> limit what styles you can have at a given level.
>

**Comment by Kostronor** [comment]
If a style only has a lvl 1 builder but lvl10 townhall, what does the person do? Builder would stop building townhalls.

**Comment by Raycoms** [comment]
If builder >= requestedTownhallLevel || builder == maxLevel
DoIt

2016-10-10 4:13 GMT-03:00 Marvin Hofmann <notifications@github.com>:

**Comment by cltnschlosser** [comment]
Seems like you could cheap out by building using styles with lower max levels.

**Comment by DarVillous** [comment]
Not to play devils advocate from beyond, but...
Doesn't "cheaping out" fall into a play style choice?  By allowing custom
styles, haven't we already opened that possibility?  And requiring a fixed
number of levels for a building will cause issues if the style only has 3
levels instead 5...or 6..or x

I recall having discussions with Ray about setting the max benefit of
certain buildings and scaling it based on the number of levels as well as
the possibility of a fixed bonus per level.

There is a bit of work to be done to accomplish it, but I had started doing
that once upon a time.  There were a couple of methods to refactor and some
new properties to be added while building up the styles during the load.
But it wasn't horrible.

To that point, I would modify Rays "if" slightly

If builder >= requestedHutLevel || builder == maxBuilderLevelAcrossAllStyles
DoIt

This way, they cant build a builder in a 3 level style set and build the
level 10 Townhall from the uber castle in the sky style without building
another builder.
